x/concentrated-liquidity/model/pool_test.go

```diff
@@ -347,3 +348,132 @@ func (s *ConcentratedPoolTestSuite) TestNewConcentratedLiquidityPool() {
 		})
 	}
 }
+
+func (suite *ConcentratedPoolTestSuite) TestCalcActualAmounts() {
+	var (
+		tickToSqrtPrice = func(tick int64) sdk.Dec {
+			sqrtPrice, err := math.TickToSqrtPrice(sdk.NewInt(tick))
+			suite.Require().NoError(err)
+			return sqrtPrice
+		}
+
+		defaultLiquidityDelta = sdk.NewDec(1000)
+
+		lowerTick      = int64(-99)
+		lowerSqrtPrice = tickToSqrtPrice(lowerTick)
+
+		midtick      = int64(2)
+		midSqrtPrice = tickToSqrtPrice(midtick)
+
+		uppertick      = int64(74)
+		upperSqrtPrice = tickToSqrtPrice(uppertick)
+	)
+
+	tests := map[string]struct {
+		currentTick    int64
+		lowerTick      int64
+		upperTick      int64
+		liquidityDelta sdk.Dec
+		expectError    error
+
+		expectedAmount0 sdk.Dec
+		expectedAmount1 sdk.Dec
+	}{
+		"current in range, positive liquidity": {
+			currentTick:    midtick,
+			lowerTick:      lowerTick,
+			upperTick:      uppertick,
+			liquidityDelta: defaultLiquidityDelta,
+
+			expectedAmount0: math.CalcAmount0Delta(defaultLiquidityDelta, midSqrtPrice, upperSqrtPrice, true),
+			expectedAmount1: math.CalcAmount1Delta(defaultLiquidityDelta, midSqrtPrice, lowerSqrtPrice, true),
+		},
+		"current in range, negative liquidity": {
+			currentTick:    midtick,
+			lowerTick:      lowerTick,
+			upperTick:      uppertick,
+			liquidityDelta: defaultLiquidityDelta.Neg(),
+
+			expectedAmount0: math.CalcAmount0Delta(defaultLiquidityDelta.Neg(), midSqrtPrice, upperSqrtPrice, false),
+			expectedAmount1: math.CalcAmount1Delta(defaultLiquidityDelta.Neg(), midSqrtPrice, lowerSqrtPrice, false),
+		},
```

Comment:
Maybe out of the scope of this PR/test, but I would like to see a functional test that compares the CalcActualAmount of a positive liquidity to the same negative liquidity and ensure it is properly rounding as expected (when compared to each other)

Reply:
done : [c338135](https://github.com/osmosis-labs/osmosis/pull/5005/commits/c33813505e517c610b2d13c38d331a7492147317)